tools/shared/src/main/scala/org/scalajs/core/tools/javascript/ScalaJSClassEmitter.scala

```diff
@@ -487,12 +487,41 @@ final class ScalaJSClassEmitter(semantics: Semantics, outputMode: OutputMode,
 
     val createAccessor = {
       val moduleInstanceVar = envField("n", className)
+
+      val assignModule = {
+        moduleInstanceVar :=
+          js.Apply(js.New(encodeClassVar(className), Nil) DOT js.Ident("init___"), Nil)
+      }
+
+      val initBlock = semantics.moduleInit match {
+        case CheckedBehavior.Unchecked =>
+          js.If(!(moduleInstanceVar), assignModule, js.Skip())
+        case CheckedBehavior.Compliant =>
+          js.If(moduleInstanceVar === js.Undefined(),
+            js.Block(
+                moduleInstanceVar := js.Null(),
+                assignModule
+            ),
+            js.Skip())
+        case CheckedBehavior.Fatal =>
+          js.If(moduleInstanceVar === js.Undefined(), {
+            js.Block(
+                moduleInstanceVar := js.Null(),
+                assignModule
+            )
+          }, js.If(moduleInstanceVar === js.Null(), {
+            // throw new UndefinedBehviorError(
+            //     "Initializer of $className called before completion")
+            val decodedName = Definitions.decodeClassName(className).stripSuffix("$")
+            val msg = s"Initializer of $decodedName called twice"
```

Comment:
The text does not match the one in the comment. I prefer the one in the comment.